### PR TITLE
Adds a software definition for datadog-trace-agent

### PR DESCRIPTION
**Pull request by talwai**
In plain English, the build steps are:
1. Download and install go1.7.1
2. Checkout github.com/DataDog/datadog-trace-agent
3. Download and install datadog-trace-agent build deps
4. Build the datadog-trace-agent with go1.7.1
5. Drop the compiled binary into `install_dir/bin/trace-agent`

This software definition has been tested on builds for:
- [x] deb x86_64
- [ ] rpm x86_64

### CODE REVIEW

**Comment by remh** [comment]
Let's not include it for i386 versions of the agent.

We are working on deprecating those packages anyway.

File: config/software/datadog-trace-agent.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+build do
+   ship_license "https://raw.githubusercontent.com/DataDog/datadog-trace-agent/#{version}/LICENSE"
+   # download go1.7
+   command "curl #{gourl} -o #{goout}"
+   command "mkdir -p #{godir}"
+   command "tar zxfv #{goout} -C #{godir}"
+
+   # Download datadog-trace-agent without installing
+   command "git clone https://github.com/DataDog/datadog-trace-agent.git $GOPATH/src/github.com/DataDog/datadog-trace-agent", :env => env
```
Comment (remh) [inline config/software/datadog-trace-agent.rb]: you shouldn't clone that manually but use the `source` helper.

e.g.:
https://github.com/DataDog/dd-agent-omnibus/blob/master/config/projects/datadog-agent.rb#L23

It's way cleaner and easier to get the right version

File: config/software/datadog-trace-agent.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   # Checkout datadog-trace-agent's build dependencies
+   command "#{gobin} get -d github.com/robfig/glock", :env => env
+
+   # Pin build deps to known versions
+   command "git checkout 7bc8ce51048e2adc11733f90a87b1c02fb7feebe", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/robfig/glock"
+   command "#{gobin} install github.com/robfig/glock", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/robfig/glock"
+
+   # Checkout and build datadog-trace-agent
+   command "$GOPATH/bin/glock sync github.com/DataDog/datadog-trace-agent", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/DataDog/datadog-trace-agent"
+   command "git checkout master && git pull", :env => env, :cwd => "#{Omnibus::Config.cache_dir}/src/datadog-trace-agent/src/github.com/DataDog/datadog-trace-agent"
```
Comment (remh) [inline config/software/datadog-trace-agent.rb]: i don't think you should always use master and you shouldn't do the checkout yourself you should let omnibus does that for you and have a proper tagging scheme in place for the datadog-trace-agent repo (similar to the dd-agent one)

**Comment by remh** [comment]
looks better
i also think we should move it to https://github.com/DataDog/dd-agent-omnibus/tree/master/config/software

Our core apps software definitions should be on this repo and we should keep omnibus-software for our dependencies.

**Comment by talwai** [comment]
Sounds good, I will close this and move the same definition over to `dd-agent-omnibus`

**Comment by talwai** [comment]
@remh superseded by https://github.com/DataDog/dd-agent-omnibus/pull/107
